src/test/java/net/cryptic_game/microservice/network/endpoint/NetworkEndpointTest.java

```diff
@@ -0,0 +1,354 @@
+        });
+
+        JSON data = new JSON(JSONBuilder.anJSON().add("source", source.toString()).add("destination", destination.toString()).build());
+
+        assertEquals(JSONBuilder.anJSON().add("connected", true).build(), NetworkEndpoint.check(data, "ms"));
+
+        members.remove(1);
+        assertEquals(JSONBuilder.anJSON().add("connected", false).build(), NetworkEndpoint.check(data, "ms"));
+    }
+
+    @Test
+    public void membersTest() {
+        networks = new ArrayList<>();
+        members = new ArrayList<>();
+
+        UUID network = UUID.randomUUID();
+
+        networks.add(new Network(network, UUID.randomUUID(), UUID.randomUUID(), rand.nextBoolean(), "network_" + rand.nextInt(100)));
+
+        members.add(new Member(UUID.randomUUID(), UUID.randomUUID(), UUID.randomUUID(), network));
+        members.add(new Member(UUID.randomUUID(), UUID.randomUUID(), UUID.randomUUID(), network));
+
+        PowerMockito.when(Network.get(Mockito.any())).thenAnswer((InvocationOnMock invocationOnMock) -> {
+            for (Network all : networks) {
+                if (all.getUUID().equals(invocationOnMock.getArgument(0))) {
+                    return all;
+                }
+            }
+
+            return null;
+        });
+
+        PowerMockito.when(Member.getMembers(Mockito.any())).thenAnswer((InvocationOnMock invocationOnMock) -> {
+            List<Member> returnMembers = new ArrayList<>();
+            for (Member all : members) {
+                if (all.getNetwork().equals(invocationOnMock.getArgument(0))) {
+                    returnMembers.add(all);
+                }
+            }
+
+            return returnMembers;
+        });
+
+        PowerMockito.when(Device.checkPermissions(Mockito.any(), Mockito.any())).thenReturn(true);
+
+        JSON data = new JSON(JSONBuilder.anJSON().add("uuid", network.toString()).build());
+
+        JSONObject json = NetworkEndpoint.members(data, UUID.randomUUID());
+
+        assertTrue(json.containsKey("members"));
+        List<JSONObject> memberList = (List<JSONObject>) json.get("members");
+
+        assertEquals(2, memberList.size());
+
+        members.clear();
+
+        json = NetworkEndpoint.members(data, UUID.randomUUID());
+
+        assertTrue(json.containsKey("members"));
+        memberList = (List<JSONObject>) json.get("members");
+
+        assertEquals(0, memberList.size());
+
+        PowerMockito.when(Device.checkPermissions(Mockito.any(), Mockito.any())).thenReturn(false);
+        assertEquals(JSONBuilder.anJSON().add("error", "network_not_found").build(), NetworkEndpoint.members(data, UUID.randomUUID()));
+
+        PowerMockito.when(Device.checkPermissions(Mockito.any(), Mockito.any())).thenReturn(true);
+        networks.clear();
+
+        assertEquals(JSONBuilder.anJSON().add("error", "network_not_found").build(), NetworkEndpoint.members(data, UUID.randomUUID()));
+    }
+
+    @Test
+    public void deleteUserTest() throws Exception {
```

Comment:
Why? `throws Exception`

Reply:
This is needed because we use PowerMockito